drivers/soundwire/intel_init.c

```diff
@@ -363,6 +364,34 @@ void sdw_intel_exit(struct sdw_intel_ctx *ctx)
 }
 EXPORT_SYMBOL_NS(sdw_intel_exit, SOUNDWIRE_INTEL_INIT);
 
+void sdw_intel_pre_exit(struct sdw_intel_ctx *ctx)
+{
+	struct sdw_intel_link_res *link;
+
+	list_for_each_entry(link, &ctx->link_list, list) {
+		struct sdw_bus *bus = &link->cdns->bus;
+		struct sdw_slave *slave;
+		unsigned long time;
+
+		list_for_each_entry(slave, &bus->slaves, node) {
+			/*
+			 * We usually do regcache_sync in peripherals runtime resume function.
+			 * But SDW interrupt will be disable in the exit procedure and
+			 * peripherals will no longer access SDW registers. Disable peripherals
+			 * pm runtime to avoid prevent peripherals from accessing SDW registers.
+			 */
+			pm_runtime_disable(&slave->dev);
```

Comment:
sorry @bardliao I am having hard time reconciling this PR with existing comments and code:

Intel aux device says pm_runtime is already disabled?
https://github.com/thesofproject/linux/blob/e915e8b9a14fd32332fd00879eb13c689942c72e/drivers/soundwire/intel_auxdevice.c#L395


and the master code is also with pm_runtime disabled.
https://github.com/thesofproject/linux/blob/e915e8b9a14fd32332fd00879eb13c689942c72e/drivers/soundwire/master.c#L184

Reply:
@plbossart The PR is trying to disable `peripherals` pm runtime, not the `managers`. [pm_request_resume(&slave->dev);](https://github.com/thesofproject/linux/blob/topic/sof-dev/drivers/soundwire/bus.c#L1981) is called right after a peripheral is initialized, and this PR is trying to avoid it since the runtime resume function will do `regcache_sync()`.
Here are some logs for reference.
1. [dmesg.txt](https://github.com/thesofproject/linux/files/14399133/dmesg.txt): add trace [0001-Add-trace.patch](https://github.com/thesofproject/linux/files/14399154/0001-Add-trace.patch), without this PR
2. [dmesg_w4838.txt](https://github.com/thesofproject/linux/files/14399134/dmesg_w4838.txt) add trace and with thie PR
3. [dmesg_disable_m_pm.txt](https://github.com/thesofproject/linux/files/14399136/dmesg_disable_m_pm.txt) replace `pm_runtime_disable(&slave->dev);` by `pm_runtime_disable(&bus->md->dev);` in `sdw_intel_pre_exit()`
```patch
diff --git a/drivers/soundwire/intel_init.c b/drivers/soundwire/intel_init.c
index 3baa06297c34..528ac11a4dc3 100644
--- a/drivers/soundwire/intel_init.c
+++ b/drivers/soundwire/intel_init.c
@@ -373,6 +373,8 @@ void sdw_intel_pre_exit(struct sdw_intel_ctx *ctx)
                struct sdw_slave *slave;
                unsigned long time;

+               dev_info(&bus->md->dev, "bard: %s calling pm_runtime_disable\n", __func__);
+               pm_runtime_disable(&bus->md->dev);
                list_for_each_entry(slave, &bus->slaves, node) {
                        /*
                         * We usually do regcache_sync in peripherals runtime resume function.
@@ -380,7 +382,8 @@ void sdw_intel_pre_exit(struct sdw_intel_ctx *ctx)
                         * peripherals will no longer access SDW registers. Disable peripherals
                         * pm runtime to avoid prevent peripherals from accessing SDW registers.
                         */
-                       pm_runtime_disable(&slave->dev);
+//                     dev_info(&slave->dev, "bard: %s calling pm_runtime_disable\n", __func__);
+//                     pm_runtime_disable(&slave->dev);
                        /*
                         * We need to wait for peripherals to be initialized before disabling
                         * SDW interrupt
```

Reply:
Changes V2:
Instead of disabling runtime pm, using pm_runtime_resume(&slave->dev); to wait for peripherals being initialized and regmap synced.